src/main/java/levy/art/RijkSearchFrame.java

```diff
@@ -0,0 +1,131 @@
+package levy.art;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URL;
+
+import com.andrewoid.ApiKey;
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import levy.art.json.ArtObject;
+import levy.art.json.RijksCollection;
+
+
+public class RijkSearchFrame extends JFrame {
+    private JTextField searchField;
+    private JButton prevButton;
+    private JButton nextButton;
+    private JPanel imagesPanel;
+    private int currentPage = 1;
+    private RijkService rijkService;
+    private Disposable disposable;
+    private String apiKey;
+
+    public RijkSearchFrame(RijkService rijkService) {
+        this.rijkService = rijkService;
+
+        ApiKey apiKey = new ApiKey();
+        this.apiKey = apiKey.get();
+
+        setTitle("Rijks");
+        setSize(1000, 800);
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        searchField = new JTextField(20);
+        prevButton = new JButton("Previous Page");
+        nextButton = new JButton("Next Page");
+        JPanel topPanel = new JPanel();
+        topPanel.add(prevButton);
+        topPanel.add(searchField);
+        topPanel.add(nextButton);
+        add(topPanel, BorderLayout.NORTH);
+
+        imagesPanel = new JPanel(new GridLayout(3, 4, 10, 10));
+        add(new JScrollPane(imagesPanel), BorderLayout.CENTER);
+
+        prevButton.addActionListener(e -> {
+            if (currentPage > 1) {
+                currentPage--;
+                searchArt();
+            }
+        });
+
+        nextButton.addActionListener(e -> {
+            currentPage++;
+            searchArt();
+        });
+
+        searchField.addActionListener(e -> {
+            currentPage = 1;
+            searchArt();
+        });
+
+        searchArt();
+    }
+
+    private void searchArt() {
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
+        }
+
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
```

Comment:
Do this one time. Not each time this method is called. You have a member variable for this.

Reply:
> Do this one time. Not each time this method is called. You have a member variable for this.

You are still doing this every time you search.

Reply:
should i only be doing it once in my service also?

Reply:
should i use a @beforeAll

Reply:
> should i use a @beforeAll

That's only for tests

Reply:
> should i only be doing it once in my service also? 

Once in you **application**

Reply:
so i shoudl set it once like in my collection and never have to use that usage again?

Reply:
i put it just in my test where else should it not be?

Reply:
The way you had it in your test was fine. I never asked you to change that.

The way you have it in your application now is correct.